### PR TITLE
VACMS-8327: Install HTML tag usage module to audit <em> tags.

### LINKED ISSUE #8327: Audit italics usage in rich text editors
See #8319 for background. 

## User Story or Problem Statement

Before we remove support for italics in the two rich text editors, we need to understand what impact this will have on existing content and content editors, so that we can plan a change management strategy for this change. 

## Acceptance Criteria
- [x] We have a tool for analysing use of italics on CMS-supported pages. it doesn't have to be a drupal view, could be a crawl of www.va.gov sitemap. 
  - [x] the tool provides an export of results
  - [x]  it would be nice if the results did not appear in a modal, but on a full page.
  - [x] stretch the tool can filter results by html tag type
- [x] any patterns that may help our change management team think through the impact of this change for editors (for example, it's mostly on VAMC Detail pages, or it's mostly added by a certain editor) 

## Implementation steps
TBD 

## Design principles

Veteran-centered
- [ ] `Single source of truth`: Increase reliability and consistency of content on VA.gov by providing a single source of truth.
- [x] `Accessible, plain language`: Provide guardrails and guidelines to ensure content quality.
- [ ] `Purposely structured content`: Ensure Content API can deliver content whose meaning matches its structure.
- [ ] `Content lifecycle governance`: Produce tools, processes and policies to maintain content quality throughout its lifecycle.

Editor-centered
- [ ] `Purpose-driven`: Create an opportunity to involve the editor community in VA’s mission and content strategy goals.
- [ ] `Efficient`: Remove distractions and create clear, straightforward paths to get the job done.
- [ ] `Approachable`: Offer friendly guidance over authoritative instruction.
- [ ] `Consistent`: Reduce user’s mental load by allowing them to fall back on pattern recognition to complete tasks.
- [ ] `Empowering`: Provide clear information to help editors make decisions about their work.

### CMS Team
Please check the team(s) that will do this work. 

- [ ] `CMS Program`
- [ ] `Platform CMS Team`
- [ ] `Sitewide CMS Team ` (leave Sitewide unchecked and check the specific team instead)
  - [x] `⭐️ Content ops`
  - [ ] `⭐️ CMS experience`
  - [ ] `⭐️ Offices`
  - [ ] `⭐️ Product support`
  - [ ] `⭐️ User support`

### PR DESCRIPTION
**Pull request by msbtterswrth**
## Description

Closes #8327 

## Screenshots
image

## QA steps

What needs to be checked to prove this works?  
What needs to be checked to prove it didn't break any related things?  
What variations of circumstances (users, actions, values) need to be checked?  

As cms admin
1. I can got to the HTML usage audit (`admin/reports/html_tag_usage`) and
   - [x] Validate that the audit runs
2. Then click on the `<em>` count column and
   - [x] Validate that the new page shows me instances of `<em>` usage
3. Then validate Acceptance Criteria from issue
   - [x] the tool provides an export of results
   - [ ] it would be nice if the results did not appear in a modal, but on a full page.
   - [x] (stretch) the tool can filter results by html tag type

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `Platform CMS Team`
- [ ] `Sitewide CMS Team`
  - [x] `⭐️ Content ops`
  - [ ] `⭐️ CMS Experience`
  - [ ] `⭐️ Offices`
  - [ ] `⭐️ Product Support`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [ ] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [ ] No announcement is needed for this code change.
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

**Comment by ethanteague** [comment]
Filters and csv
image

**Comment by swirtSJW** [comment]
A beta module with a notice like this is a bit concerning
image

**Comment by ethanteague** [comment]
> A beta module with a notice like this is a bit concerning image

Yeah. Wonder if we should add an environment check to enable / disable, @swirtSJW ?

File: config/sync/core.extension.yml [hunk 1]
```diff
@@ -21,6 +21,7 @@ module:
   breakpoint: 0
   cer: 0
   ckeditor: 0
+  ckeditor5: 0
```
Comment (swirtSJW) [inline config/sync/core.extension.yml:24]: This seems unexpected.

File: docroot/modules/custom/va_gov_db/va_gov_db.install [hunk 2]
```diff
@@ -798,3 +798,27 @@ function va_gov_db_update_9002(&$sandbox) {
 
   return $messages;
 }
+
+/**
+ * Fix field_error_message on paragraph to add text format when value is NULL.
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_db/va_gov_db.install:803]: This is misleading.  It is not fixing the value when something is null.   It is forcing a value in for a specific entity id.

**Review by swirtSJW — CHANGES_REQUESTED** [review]
A couple niggly bits.  

@msbtterswrth  @ethanteague  @kevwalsh 
On the bigger picture of this being not recommended for production and in beta.    Do we need this enabled on prod staging and dev?

Can we just enable this on local or tugboat when needed to run the report?

File: docroot/modules/custom/va_gov_db/va_gov_db.install (hunk 2 above)
Comment (ethanteague) [reply docroot/modules/custom/va_gov_db/va_gov_db.install:803]: Dangit - I _knew_ this was a bad message, but I got lazy and didn't clean it up, @swirtSJW

File: docroot/modules/custom/va_gov_db/va_gov_db.install [hunk 3]
```diff
@@ -798,3 +798,27 @@ function va_gov_db_update_9002(&$sandbox) {
...
+  \Drupal::logger('va_gov_db')->info('Update field_error_message to full_html');
+  \Drupal::database()
+    ->update('paragraph__field_error_message')
+    ->condition('entity_id', 4102)
+    ->fields([
+      'field_error_message_format' => 'rich_text',
+    ])
+    ->execute();
+
+  \Drupal::logger('va_gov_db')->info('Update field_error_message revision to full_html');
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_db/va_gov_db.install:816]: Same here.  Should come after.

File: docroot/modules/custom/va_gov_db/va_gov_db.install [hunk 4]
```diff
@@ -798,3 +798,27 @@ function va_gov_db_update_9002(&$sandbox) {
 
   return $messages;
 }
+
+/**
+ * Fix field_error_message on paragraph to add text format when value is NULL.
+ */
+function va_gov_db_update_9003() {
+
+  \Drupal::logger('va_gov_db')->info('Update field_error_message to full_html');
```
[the remark this answers is not in the document]
Comment (ethanteague) [reply docroot/modules/custom/va_gov_db/va_gov_db.install:807]: Yeah.....not sure why I did this. Will fix

File: config/sync/core.extension.yml (hunk 1 above)
Comment (swirtSJW) [reply config/sync/core.extension.yml:24]: This is still here.

**Review by swirtSJW — CHANGES_REQUESTED** [review]

File: composer.json [hunk 5]
```diff
@@ -413,7 +413,7 @@
                 "3145496 - Make d9 compatible": "https://www.drupal.org/files/issues/2020-06-05/3145496-6.patch"
             },
             "drupal/html_tag_usage": {
-                "3270220 - Add csv generator": "https://www.drupal.org/files/issues/2022-03-17/csv-generator-for-tag-result-sets.patch"
+                "3270450 - Add csv generator and filters": "https://www.drupal.org/files/issues/2022-03-21/add-filters-and-csv-5.patch"
```
Comment (swirtSJW) [inline composer.json:416]: Link to this issue on upstream module https://www.drupal.org/project/html_tag_usage/issues/3270450

Comment (swirtSJW) [reply composer.json:416]: Nice work on this patch.

**Review by swirtSJW — APPROVED** [review]
Nice work.

**Pull request merged**